tests/phpunit/test-emitter.php

```diff
@@ -623,4 +623,55 @@ public function test_surrogate_keys_with_filter_override() {
 		remove_all_filters("pantheon_should_add_terms");
 	}
 
+	/**
+	 * Test that queries on archive pages with multiple post types doesn't throw an error.
+	 *
+	 * @see https://getpantheon.atlassian.net/browse/BUGS-6934
+	 */
+	public function test_surrogate_keys_multiple_post_types() {
+		global $wp_query;
+
+		// Set up posts of different post types.
+		$this->factory->post->create( [
+			'post_type' => 'post',
+			'post_title' => 'test product'
+		] );
+		$this->factory->post->create( [
+			'post_type' => 'post',
+			'post_title' => 'test post'
+		] );
+
+		// Query the posts we just created.
+		$wp_query = new WP_Query( [
+			's' => 'test',
+			'post_type' => [ 'post', 'product' ],
+		] );
+		$wp_query->is_archive = true;
+		$wp_query->is_post_type_archive = true;
+
+		// Ensure that we don't get a WP Error from the surrogate key generator. This is a bit superfluous since if there was an error, we'd see it in the tests.
+		$this->assertTrue( ! is_wp_error( Emitter::get_main_query_surrogate_keys() ) );
+
+		// Test that we have surrogate keys for the post and product we created.
+		if ( is_multisite() ) {
+			$this->assertContains(
+				'blog-1-post-archive',
+				Emitter::get_main_query_surrogate_keys()
+			);
+			$this->assertContains(
+				'blog-1-product-archive',
+				Emitter::get_main_query_surrogate_keys()
+			);
+		} else {
+			$this->assertContains(
+				'post-archive',
+				Emitter::get_main_query_surrogate_keys()
+			);
+			$this->assertContains(
+				'product-archive',
+				Emitter::get_main_query_surrogate_keys()
+			);
+		}
```

Comment:
```suggestion
		$wpms_prefix="";
		if ( is_multisite() ) {
			$wpms_prefix="blog-";
		}
		$post_archive_key=$wpms_prefix.'post-archive';
		$this->assertContains(
			$post_archive_key,
			Emitter::get_main_query_surrogate_keys()
		);
		$product_archive_key=$wpms_prefix.product-archive';
		$this->assertContains(
			$product_archive_key,
			Emitter::get_main_query_surrogate_keys()
		);
```
needs linting but DRYing up a bit